File: Zoo.java
import java.time.Period;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;
import java.time.LocalDate;
import java.util.Scanner;
public class Zoo {
    static Scanner sc = new Scanner(System.in);
    private ArrayList<Employee> employees = new ArrayList<>();
    private ArrayList<Animal> animals = new ArrayList<>();
    private ArrayList<Client> clients = new ArrayList<>();
    private ArrayList<Order> orders = new ArrayList<>();
    private ArrayList<Visit> visits = new ArrayList<>();

    public void registerEmployee() {
        String lastName = "";
        DateTimeFormatter formato = DateTimeFormatter.ofPattern("dd-MM-yyyy");

        System.out.println("\n\t== HA SELECCIONADO REGISTRAR UN EMPLEADO ==\n");
        System.out.println("|| A continuacion ingresara los datos del nuevo empleado ||");
        System.out.println("Ingrese su nombre completo (Un nombre seguido de sus apellidos): ");
        String name = sc.nextLine();

        for (int i = 0; i < name.length(); i++) {
            if (name.charAt(i) == ' ') {
                lastName = name.substring(i + 1);
                name = name.substring(0, i);
                break;
            }
        }

        System.out.println("Ingrese el cumpleaños del empleado (dd-MM-yyyy)");
        String birthday = sc.nextLine();

        int age = Period.between(LocalDate.parse(birthday, formato), LocalDate.now()).getYears();
        if (age < 18) {
            System.out.println("El empleado es menor de edad!! LLAMANDO AL 911");
            return;
        }

        System.out.println("Ingrese el RFC del empleado: ");
        String rfc = sc.nextLine();
        System.out.println("Ingrese el CURP del empleado: ");
        String curp = sc.nextLine();
        System.out.println("Ingrse el salario que recibira el empleado: ");
        double salary = sc.nextDouble();
        System.out.println("Ingrese el horario en el que el empleado trabajara (XX:XX - XX:XX)");
        String schedule = sc.nextLine();
        schedule = sc.nextLine();
        String availableRoles = "administracionguiamantenimientoveterinario";
        System.out.println("Que rol desempeñara el empleado (Administracion, Guia, Mantenimiento, Veterinario)");
        String role = sc.nextLine();
        while (!availableRoles.contains(role.toLowerCase())) {

            System.out.println("Rol no reconocido por el sistema");
            System.out.println("Que rol desempeñara el empleado (Administracion, Guia, Mantenimiento, Veterinario)");
            role = sc.nextLine();
        }

        Employee employee = new Employee(name, lastName, birthday, rfc, curp, salary, schedule, role, age);
        employees.add(employee);
        System.out.println("\n\t== SE HA REGISTRADO A " + name + " CORRECTAMENTE ==\n");
    }

    public void registerAnimal() {
        ArrayList<String> diseases = new ArrayList<>();
        boolean isVaccinated;

        System.out.println("\n\t== HA SELECCIONADO REGSITRAR UN ANIMAL ==\n");
        System.out.println("|| A continuacion ingresera los datos del animal ||");
        System.out.println("Que tipo de animal es: ");
        String animalType = sc.nextLine();
        System.out.println("Cual es la fecha de nacimiento: (dd-MM-yyyy)");
        String birthday = sc.nextLine();
        System.out.println("Ingrese el peso del animal: ");
        double weight = sc.nextDouble();
        System.out.println("El animal cuenta con efermedades: (S/N)");
        char isSick = sc.next().charAt(0);

        while ((isSick == 'S' || isSick == 's')) {
            System.out.println("Con que enfermedad cuenta el animal: ");
            String disease = sc.next();
            diseases.add(disease);
            System.out.println("Cuenta con mas enfermedades: (S/N)");
            isSick = sc.next().charAt(0);
        }

        System.out.println("Cada cuantas horas se alimentara al animal: ");
        int feedingSchedule = sc.nextInt();
        System.out.println("Ingrese su tipo de alimentacion: (Herbivoro, Carnivoro, Omnivoro)");
        String foodType = sc.nextLine();
        foodType = sc.nextLine();
        System.out.println("El animal esta vacunado: ");
        char Vaccinated = sc.next().charAt(0);
        isVaccinated = Vaccinated == 'S' || Vaccinated == 's';

        Animal animal = new Animal(animalType, birthday, weight, diseases, feedingSchedule, foodType, isVaccinated);
        animals.add(animal);
        System.out.println("\n\t== SE HA REGISTRADO EL ANIMAL ==\n");
    }

    public void registerClient() {
        String lastName = "";
        DateTimeFormatter formato = DateTimeFormatter.ofPattern("dd-MM-yyyy");

        System.out.println("\n\t== HA SELECCIONADO REGISTRAR UN CLIENTE ==\n");
        System.out.println("|| A continuacion ingresara los datos del nuevo cliente ||");
        System.out.println("Ingrese su nombre completo (Un nombre seguido de sus apellidos): ");
        String name = sc.nextLine();

        for (int i = 0; i < name.length(); i++) {
            if (name.charAt(i) == ' ') {
                lastName = name.substring(i + 1);
                name = name.substring(0, i);
                break;
            }
        }

        System.out.println("Ingrese el cumpleaños del cliente (dd-MM-yyyy)");
        String birthday = sc.nextLine();
        int age = Period.between(LocalDate.parse(birthday, formato), LocalDate.now()).getYears();
        System.out.println("Ingrese el CURP del cliente: ");
        String curp = sc.nextLine();

        Client client = new Client(name, lastName, birthday, curp, age);
        clients.add(client);
        System.out.println("\n\t== SE HA REGISTRADO A " + name + " CORRECTAMENTE ==\n");
    }

    public void registerVisit() {
        ArrayList<String> visitants = new ArrayList<String>();
        String employee;
        String clientName;
        int adults = 0;
        int children = 0;
        double totalPrice = 0;
        ArrayList<String> clientsInVisit = new ArrayList<>();

        System.out.println("\n\t== HA SELECCIONADO REGISTRAR UNA VISITA =\n");
        System.out.println("|| A continuacion ingresara los datos de la visita ||");
        printEmployees("Guia");
        do {
            System.out.println("\nQuien estara a cargo de esta visita: (Seleccione el ID del empleado)");
            String inCharge = sc.next();
            employee = FindID.employeeID(this.employees, inCharge);
        } while (employee == "incorrect");
        printClients();
        System.out.println("\nA continuacion ingrese los visitantes que entraran a la visita: (Ingrese el ID de los visitantes)\n");
        String client = sc.next();

        do {
            clientName = FindID.clientName(clients, client);
            if (clientName.equalsIgnoreCase("incorrect") || clientsInVisit.contains(client)) {
                System.out.println("ID Inexistente o ya ingresado. Ingrese otro ID: ");
                client = sc.next();
            } else {
                System.out.println("\nCliente " + clientName + " se ha agregado a la visita");
                visitants.add(FindID.clientName(clients, client));
                clientsInVisit.add(client);

                if (FindID.clientAge(clients, client) >= 18) {
                    if (FindID.clientVisists(clients, client) >= 5)
                        totalPrice += 80;
                    else {
                        totalPrice += 100;
                    }
                    adults++;
                } else {
                    if (FindID.clientVisists(clients, client) >= 5)
                        totalPrice += 40;
                    else {
                        totalPrice += 50;
                    }
                    children++;
                }

                FindID.client(clients, client).addVisit();

                System.out.println("Si desea agregar otro visitante, ingrese el numero de otro visitante, sino, ingrese -1");
                client = sc.next();
            }
        } while (!client.equalsIgnoreCase("-1"));

        Visit visit = new Visit(employee, visitants, totalPrice, adults, children);
        visits.add(visit);
        System.out.println("\n\t==LA VISITA HA SIDO REGISTRADA EN EL SISTEMA ==\n");
    }

    public void printEmployees(String role) {
        System.out.println("\n\t== LISTA DE EMPLEADOS CON EL ROL DE " + role.toUpperCase() + " ==\n");
        for (Employee employee : employees) {
            if (employee.getRole().equalsIgnoreCase(role)) {
                System.out.printf("Empleado %s. | Nombre: %s %s | Horario: %s \n",
                        employee.getID(), employee.getName(), employee.getLastName(), employee.getHours());
            }
        }
        System.out.println();
    }

    public void printClients() {
        System.out.println("\n\t== LISTA DE VISITANTES REGISTRADOS ==");
        for (Client client : clients) {
            System.out.printf("Cliente %s. | Nombre: %s %s | Dia Nacimiento: %s | Num. Visitas: %d | Dia Registro: %s\n",
                    client.getID(), client.getName(), client.getLastName(), client.getBirthday(), client.getVists(), client.getDayRegistration());
        }
        System.out.println();
    }

    public void printEmployees() {
        System.out.println("\n\t== LISTA DE EMPLEADOS ==\n");
        for (Employee employee : employees) {
            System.out.printf("Empleado %s. | Nombre: %s %s\n",
                    employee.getID(), employee.getName(), employee.getLastName());
            System.out.printf("\tEdad: %d | Horario: %s | Rol: %s | Salario: %f \n",
                    employee.getAge(), employee.getHours(), employee.getRole(), employee.getSalary());
            System.out.printf("\tRFC: %s | CURP: %s | Dia de contratacion: %s \n",
                    employee.getRfc(), employee.getCurp(), employee.getFirstDayAsWorker());
        }
        System.out.println();
    }

    public void printAnimals() {
        System.out.println("\n\t== LISTA DE ANIMALES ==\n");
        for (Animal animal : animals) {
            String vacunado = animal.isVaccinated() ? "Si" : "No";
            System.out.printf("Animal %s. | Tipo: %s | Nacio el: %s | Esta vacunado: %s | Peso: %f | Se alimenta cada: %d horas\n",
                    animal.getID(), animal.getAnimalType(), animal.getBirthday(), vacunado, animal.getWeight(), animal.getFeedingSchedule());
            if (!animal.getDiseases().isEmpty()) {
                animal.printDiseases(animal);
            }
        }
        System.out.println();
    }

    public void printVisits() {
        System.out.println("\n\t== LISTA DE VISITAS ==\n");
        for (Visit visit : visits) {
            System.out.println("\t= VISITA " + visit.getID() + " =");
            System.out.printf("Empleado a Cargo: %s | Precio total: %f | Dia: %s\n",
                    visit.getEmployeeInCharge(), visit.getTotalPrice(), visit.getDayOfVisit());
            visit.printVisitants();
            System.out.println();
        }
    }

    public void printOrders() {
        System.out.println("\n\t== LISTA DE ORDENES ==\n");
        for (Order order : orders){
            System.out.println("\t= ORDEN PARA "+order.getEmployee().toUpperCase()+" =");
            System.out.printf("Proceso realizado: %s Fecha: %s Animal: %s \n",
                    order.getProcess(),order.getDate(),order.getAnimalID());
            if (order.getObservations() != null){
                System.out.println("Observaciones: "+order.getObservations());
            }
        }
    }

    public void orderMaintenance() {
        String employee;
        String animalId;

        printEmployees("mantenimiento");

        while (true) {
            System.out.println("Que empleado desea asignar: (Ingrese el ID)");
            String id = sc.nextLine();
            employee = FindID.employeeID(employees, id);
            if (!employee.equalsIgnoreCase("incorrect")) {
                break;
            }
        }

        printAnimals();

        do {
            System.out.println("A que animal se le hara el mantenimiento: ");
            animalId = sc.nextLine();
        } while (!FindID.animalExists(animals, animalId));

        System.out.println("Que proceso realizara el empleado (Alimentar, Limpiar, Cuidar)");
        String process = sc.nextLine();
        System.out.println("El empleado desea realizar observaciones: (S/N)");
        char ob = sc.nextLine().charAt(0);
        if (ob == 'S' || ob == 's') {
            System.out.println("Que observaciones tuvo el empleado: ");
            String observation = sc.nextLine();
            Order order = new Order(employee, animalId, process, observation);
            orders.add(order);
        } else {
            Order order = new Order(employee, animalId, process);
            orders.add(order);
        }
        System.out.println("\n\t== SE HA REGISTRADO LA ORDEN CORRECTAMENTE ==\n");

    }

    public void modifyClient(){
        boolean flag = true;
        String id;

        System.out.println("\n\t==HA SELECCIONADO MODIFICAR UN CLIENTE==\n");
        printClients();

        while(true) {
            System.out.println("\nQue cliente desea modificar: (INGRESE SU ID)");
            id = sc.nextLine();

            if (FindID.clientName(clients, id) != "incorrect"){
                System.out.println("Ha seleccionado a: " + FindID.clientName(clients, id));
                break;
            }

            else {
                System.out.println("\nID INGRESADO INCORRECTO\n");
            }
        }
        while (flag) {
            System.out.println("Que desea modificar: ");
            System.out.println("1. Nombre");
            System.out.println("2. Apellido");
            System.out.println("3. CURP");
            System.out.println("4. Num. Visitas");
            System.out.println("5. Salir");

            int opt = sc.nextInt();

            switch (opt){
                case 1 -> FindID.client(clients, id).setName();
                case 2 -> FindID.client(clients, id).setLastName();
                case 3 -> FindID.client(clients, id).setCurp();
                case 4 -> FindID.client(clients, id).setVisits();
                case 5 -> flag = false;
                default -> System.out.println("Opcion ingresada incorrecta");
            }
        }

    }

    public void modifyEmployee(){
        boolean flag = true;
        String id;

        System.out.println("\n\t==HA SELECCIONADO MODIFICAR UN EMPLEADO==\n");
        printEmployees();

        while(true) {
            System.out.println("\nQue empleado desea modificar: (INGRESE SU ID)");
            id = sc.nextLine();

            if (FindID.employeeID(employees, id) != "incorrect"){
                System.out.println("Ha seleccionado a: " + FindID.employeeID(employees, id));
                break;
            }

            else {
                System.out.println("\nID INGRESADO INCORRECTO\n");
            }
        }

        while (flag) {
            System.out.println("\nQue desea modificar: ");
            System.out.println("1. Nombre");
            System.out.println("2. Apellido");
            System.out.println("3. Rol");
            System.out.println("4. Salario");
            System.out.println("5. Horario");
            System.out.println("6. Salir");

            int opt = sc.nextInt();

            switch (opt){
                case 1 -> FindID.employee(employees, id).setName();
                case 2 -> FindID.employee(employees, id).setLastName();
                case 3 -> FindID.employee(employees, id).setRole();
                case 4 -> FindID.employee(employees, id).setSalary();
                case 5 -> FindID.employee(employees, id).setHours();
                case 6 -> flag = false;
                default -> System.out.println("Opcion ingresada incorrecta");
            }
        }
    }

    public void modifyAnimal(){
        boolean flag = true;
        String id;

        System.out.println("\n\t==HA SELECCIONADO MODIFICAR UN ANIMAL==\n");
        printAnimals();

        while(true) {
            System.out.println("\nQue animal desea modificar: (INGRESE SU ID)");
            id = sc.nextLine();

            if (FindID.animalExists(animals, id)){
                System.out.println("Ha seleccionado a el animal con ID: "+id);
                break;
            }

            else {
                System.out.println("\nID INGRESADO INCORRECTO\n");
            }
        }

        while (flag) {
            System.out.println("\nQue desea modificar: ");
            System.out.println("1. Peso");
            System.out.println("2. Dias de Alimentacion");
            System.out.println("3. Vacunar al Animal");
            System.out.println("4. Salir");

            int opt = sc.nextInt();

            switch (opt){
                case 1 -> FindID.animal(animals, id).setWeight();
                case 2 -> FindID.animal(animals, id).setFeedingSchedule();
                case 3 -> FindID.animal(animals, id).setVaccinated();
                case 4 -> flag = false;
                default -> System.out.println("Opcion ingresada incorrecta");
            }
        }
    }

    public void deleteClient(){
        String id;

        System.out.println("\n\t==HA SELECCIONADO ELIMINAR UN CLIENTE==\n");
        printClients();

        while(true) {
            System.out.println("\nQue cliente desea eliminar: (INGRESE SU ID)");
            id = sc.nextLine();

            if (FindID.clientName(clients, id) != "incorrect"){
                System.out.println("Ha seleccionado a: " + FindID.clientName(clients, id));
                break;
            }

            else {
                System.out.println("\nID INGRESADO INCORRECTO\n");
            }
        }

        System.out.println("Esta seguro? (S/N)");
        char delete = sc.nextLine().charAt(0);

        if (delete == 'S' || delete == 's'){
            System.out.printf("\nSE HA ELIMINADO A %s DEL SISTEMA\n",
                    FindID.clientName(clients, id).toUpperCase());
            clients.remove(FindID.client(clients, id));
        } else {
            System.out.println("\nNO SE ELIMINO NINGUN CLIENTE DEL SISTEMA\n");
        }
    }

    public void deleteEmployee(){
        String id;

        System.out.println("\n\t==HA SELECCIONADO ELIMINAR UN EMPLEADO==\n");
        printEmployees();

        while(true) {
            System.out.println("\nQue empleado desea eliminar: (INGRESE SU ID)");
            id = sc.nextLine();

            if (FindID.employeeID(employees, id) != "incorrect"){
                System.out.println("Ha seleccionado a: " + FindID.employeeID(employees, id));
                break;
            }

            else {
                System.out.println("\nID INGRESADO INCORRECTO\n");
            }
        }

        System.out.println("Esta seguro? (S/N)");
        char delete = sc.nextLine().charAt(0);

        if (delete == 'S' || delete == 's'){
            System.out.printf("\nSE HA ELIMINADO A %s DEL SISTEMA\n",
                    FindID.employeeID(employees, id).toUpperCase());
            employees.remove(FindID.employee(employees, id));
        } else {
            System.out.println("\nNO SE ELIMINO NINGUN EMPLEADO DEL SISTEMA\n");
        }
    }

    public void deleteAnimal(){
        String id;

        System.out.println("\n\t==HA SELECCIONADO ELIMINAR UN ANIMAL==\n");
        printAnimals();

        while(true) {
            System.out.println("\nQue animal desea eliminar: (INGRESE SU ID)");
            id = sc.nextLine();

            if (FindID.animalExists(animals, id)){
                System.out.println("Ha seleccionado a el animal con ID: "+id);
                break;
            }

            else {
                System.out.println("\nID INGRESADO INCORRECTO\n");
            }
        }

        System.out.println("Esta seguro? (S/N)");
        char delete = sc.nextLine().charAt(0);

        if (delete == 'S' || delete == 's'){
            System.out.printf("\nSE HA ELIMINADO A %s DEL SISTEMA\n",
                    FindID.animal(animals, id).getAnimalType().toUpperCase());
            animals.remove(FindID.animal(animals, id));
        } else {
            System.out.println("\nNO SE ELIMINO NINGUN ANIMAL DEL SISTEMA\n");
        }
    }

    public ArrayList<Employee> getEmployees() {
        return employees;
    }

    public ArrayList<Animal> getAnimals() {
        return animals;
    }

    public ArrayList<Client> getClients() {
        return clients;
    }

    public ArrayList<Order> getOrders() {
        return orders;
    }

    public ArrayList<Visit> getVisits() {
        return visits;
    }
}

File: Zoo.py
import Employee
import Client
import FindID
import Employee
import Animal
import Visit
import Order    

class Zoologic:
    employees=[]
    animals=[]
    clients=[]
    orders=[]
    visits=[]

    def register_employee(self):
        print("==Ha seleccionado registrar un empleado==")
        print("Ingrese el nombre del empleado (sin apellidos)")
        name=input()
        print("Ingrese el apellido del empleado")
        Last_name=input()
        print("Ingrese la fecha de nacimiento del empleado(DD-MM-YYYY)")
        Birthday=input()
        birthday=Birthday[:5]
        print("Ingrese el RFC del empleado")
        rfc=input()
        print("Ingrese la CURP del empleado")
        curp=input()
        print("Ingrese el salario del empleado")
        salary=input()
        print("Ingrese el horario del empleado(XX:XX - XX:XX)")
        schedule=input()
        rols=["administracion","Administracion","guia","Guia","mantenimiento","Mantenimiento","veterinario","Veterinario"]
        flag=True
        while flag==True:
            print("Ingrese el rol del empleado(Administracion,Guia,Mantenimiento,Veterinario)")
            rol=input()
            if rol in rols:
                rol=rol
                flag=False
            else:
                print("Rol no reconocido")

        employee=Employee.Employee(name,Last_name,curp,birthday,salary,schedule,rol,rfc)
        Zoologic.employees.append(employee)
        print("==Se ha registrado correctamente a: ",name,"==")

    def register_animal(self):
        print("==Ha seleccionado registrar un animal==")
        print("Ingrese la especie del animal: ")
        animal_Type=input()
        print("Ingrese el peso del animal: ")
        weight=input()
        print("Ingrese la fecha de nacimiento del animal(DD-MM-YYYY): ")
        birthday=input()
        print("El animal tiene enfermedades?(S/N)")
        Sick=input()
        diseases=[]
        if Sick== 'S' or Sick=='s':
            print("Con que enermedades cuenta el animal?")
            disease=input()
            diseases.append(disease)
        elif Sick=='N' or Sick=='n':
            disease="Sin enfermedades"
            diseases.append(disease)
        print("Cada cuanto de alimentara al animal?")
        feeding_schedule=input()
        types=["herbivoro","carnivoro","omnivoro"]
        flag=True
        while flag==True:
            print("Ingrese el tipo de alimentacion(herbivoro,carnivoro,omnivoro):")
            food_type=input()
            if food_type in types:
                food_type=food_type
                flag=False
            else:
                print("Tipo invalido")

        print("EL animal esta vacunado?")
        is_vaccinated=input()

        animal=Animal.Animal(animal_Type,birthday,weight,diseases,feeding_schedule,food_type,is_vaccinated)
        Zoologic.animals.append(animal)
        print("NUEVO ANIMAL REGISTRADO")

    def register_client(self):
        print("==Ha seleccionado: registrar un cliente")
        print("Ingrese el nombre del cliente (sin apellidos)")
        name= input()
        print("Ingrese los apellidos del cliente")
        Last_name= input()
        print("Ingrese la curp del cliente")
        curp=input()
        print("Ingrese el cumpleaños del cliente (DD-MM-YYYY)")
        birthday=input()
        print("Ingrese la edad del cliente")
        age=input()

        client=Client.Client(name,Last_name,curp,birthday,age)
        Zoologic.clients.append(client)

        print("Se ha registrado al cliente: ",name, "correctamente")

    def register_visit(self):
        visitants=[]
        Clients_in_visit=[]
        print("==Ha seleccionado: registrar una visita==")
        Zoologic.print_employees(self)
        flag= True
        while flag== True:
            print("Que empleado sera el guia en la visita? (ingrese el ID del empleado)")
            In_charge=input()
            employee=FindID.FindID.employeeID(Zoologic.employees,In_charge)
            if employee != "incorrect":
                break

        Zoologic.print_clients(self)
        print("A continuacion ingrese los visitantes que entraran a la visita. (ingrese el ID de los vistantes)")
        total_price=0
        adults=0
        children=0
        client=" "
        while client != "-1":
            client_name=FindID.FindID.clientName(Zoologic.clients,client)
            if client_name== "incorrect" or client in Clients_in_visit:
                print("ID inexistente o ya ingresado. Ingrese otro ID:")
                client =input()
            else:
                print("Cliente: " ,client_name, " se ha agregado a la visita")
                visitants.append(FindID.FindID.clientName(Zoologic.clients, client))
                Clients_in_visit.append(client)
                if FindID.FindID.age_valid(Zoologic.clients,client) >= 18:
                    if FindID.FindID.visits_client(Zoologic.clients,client) <=5:
                        total_price+=80
                    else:
                        total_price+=100
                    adults+=1
                else:
                    if FindID.FindID.clientVisits(Zoologic.clients,client) >=5:
                        total_price+=40
                    else:
                        total_price+=50
                    children+=1
                print("Si desea agregar otro visitante, ingrese el ID de otro visitante,sino, ingrese -1")
                client=input()
        visit= Visit.Visit(employee,visitants,total_price,adults,children)
        Zoologic.visits.append(visit)

        print("==VISITA REGISTRADA EN EL SISTEMA==")

    def print_employees(self,rol):
        print("==LISTA DE EMPLEADOS CON EL ROL DE: ", rol," ==")
        for employee in Zoologic.employees:
            if employee.rol().equalsIgnoreCase(rol):
                print("Empleado: ",employee.ID(), " Nombre: ", employee.name(),employee.Last_name(), " Horario: ", employee.schedule())
                print("==========")
    
    def print_clients(self):
        print("==LISTA DE VISITANTES REGISTRADOS==")
        for client in Zoologic.clients:
            Client.Client.Client_Info(client)
            print("==========")
            #print("Cliente: ",client, " Nombre: ",client.name(),client.Last_name()," Dia de nacimiento: ",client.birthday()," Num. de visitas: ",client.visits()," Dia de registro: ",client.registration_date())
    
    def print_employees(self):
        print("==LISTA DE EMPLEADOS==")
        for employee in Zoologic.employees:
          Employee.Employee.Employee_Info(employee)
          print("==========")
            #print("Empleado: ",employee.ID()," Nombre: ",employee.name(),employee.Last_name()," Edad: ",employee.age())
            #print("Horario: ",employee.schedule()," Rol: ",employee.rol()," Salario: ",employee.salary())
            #print(" RFC: ",employee.rfc()," CURP: ",employee.curp()," Dia de contratacion: ",employee.First_Day_As_Worker())
    
    def print_animals(self):
        print("==LISTA DE ANIMALES EN EL ZOOLOGICO==")
        for animal in Zoologic.animals:
            vacunado=Animal.Animal.is_vaccinated
            Animal.Animal.Data(animal)
            #print("Animal: ",animal.ID()," Tipo: ",animal.animal_Type()," Nacio el: ",animal.birthday())
            #print("Esta vacunado: ",vacunado," Peso: ",animal.weight(), " Se alimenta cada: ",animal.feeding_schedule())
            Animal.Animal.print_diseases(animal)
            print("==========")
    
    def print_visits(self):
        print("==LISTA DE VISITAS==")
        for visit in Zoologic.visits:
            Visit.Visit.visit_id(visit)
            Visit.Visit.visits_info(visit)
            print("==========")
           # print("Empleado a cargo: ",visit.employee_in_charge()," Precio total: ",visit.total_price()," Dia: ",visit.day_of_visit())

    def print_orders(self):
        print("==LISTA DE ORDENES==")
        for order in Zoologic.orders:
            Order.Order.orders_info(order)
            print("==========")
            #print("Empleado que ejecuto la orden: ",order.employee()," Animal al que se realizo la orden: ",order.animal_ID())
            #print("Proceso realizado: ", order.process()," Observaciones: ", order.observations()," Dia en que se realizo la orden: ",order.date())

    def create_maintenance_order(self):
        print("\n\t== HA SELECCIONADO CREAR UNA ORDEN DE MANTENIMIENTO ==\n")
        Zoologic.print_employees("mantenimiento")

        while True:
            id = input("Que empleado desea asignar: (Ingrese el ID)")
            employee = FindID.FindID.employeeID(Zoologic.employees,id)
            if employee != "incorrect":
                break
            else:
                print("ID no reconocido")

        Zoologic.print_animals(self)

        while True:
            animal_id = input("A que animal se le hara el mantenimiento: ")
            if FindID.FindID.animalExists(Zoologic.animals, animal_id):
                break
            else:
                print("ID no reconocido")

        process = input("Que proceso realizara el empleado (Alimentar, Limpiar, Cuidar)")
        ob = input("El empleado desea realizar observaciones: (S/N)")
        if ob == 'S' or ob == 's':
            observation = input("Que observaciones tuvo el empleado: ")
            order = Order.Order(employee, animal_id, process, observation)
            Zoologic.orders.append(order)
        else:
            order = Order.Order(employee, animal_id, process,"ninguna")
            Zoologic.orders.append(order)
        
        print("\n\t== SE HA REGISTRADO LA ORDEN CORRECTAMENTE ==")

    def modify_client(self):
        flag=True
        print("==Ha seleccionado: Modificar un cliente==")
        Zoologic.print_clients(self)

        while flag==True:
            print("Que cliente desea modificar? (Ingrese el ID del cliente)")
            id=input()
            if FindID.FindID.clientName(Zoologic.clients,id)!= "incorrect":
                print("Ha seleccionado a: ",FindID.FindID.clientName(Zoologic.clients,id))
            else:
                print("ID ingresado no reconocido")
            while flag:
                print("Que desea modificar ")
                print("1. Nombre ")
                print("2. Apellido")
                print("3. CURP")
                print("4. Numero de visitas")
                print("5. Salir")
                opt=int(input())

                if opt==1:
                    print("Ingrese el nombre del cliente: ")
                    Client.Client.name=input()
                   
                elif opt==2:
                    print("Ingrese el apellido del cliente: ")
                    Client.Client.Last_name=input()
                
                elif opt==3:
                    print("Ingrese la CURP del cliente: ")
                    Client.Client.curp=input()
                
                elif opt==4:
                    print("Ingrese la cantidad de visitas del cliente: ")
                    Client.Client.visits=int(input())
                
                elif opt==5:
                    flag=False
                
                else:
                    print("Opcion invalida")

    def modify_employee(self):
        flag=True
        print("==Ha seleccionado: Modificar un empleado==")
        Zoologic.print_employees(self)

        while flag==True:
            print("Que empleado desea modificar? (Ingrese el ID del empleado)")
            id=input()
            if FindID.FindID.employeeID(Zoologic.employees,id)!= "incorrect":
                print("Ha seleccionado a: ",FindID.FindID.employeeID(Zoologic.employees,id))
            else:
                print("ID ingresado no reconocido")
            while flag:
                print("Que desea modificar ")
                print("1. Nombre ")
                print("2. Apellido")
                print("3. Rol")
                print("4. Salario")
                print("5. Horario")
                print("6. Salir")
                opt=int(input())

                if opt==1:
                    print("Ingrese el nombre del empleado: ")
                    Employee.Employee.name=input()
                   
                elif opt==2:
                    print("Ingrese el apellido del empleado: ")
                    Employee.Employee.Last_name=input()
                
                elif opt==3:
                    print("Ingrese el rol del empleado: ")
                    Employee.Employee.curp=input()
                
                elif opt==4:
                    print("Ingrese el salario del empleado: ")
                    Employee.Employee.salary=input()
                
                elif opt==5:
                    print("Ingrese el horario del empleado: ")
                    Employee.Employee.schedule=input()

                elif opt==6:
                    flag=False
                
                else:
                    print("Opcion invalida")

    def modify_animal(self):
        flag=True
        print("==Ha seleccionado: Modificar un animal==")
        Zoologic.print_animals(self)

        while flag==True:
            print("Que animal desea modificar? (Ingrese el ID del animal)")
            id=input()
            if FindID.FindID.animalExists(Zoologic.animals,id)!= "incorrect":
                print("Ha seleccionado al animal con el ID: ",id)
            else:
                print("ID ingresado no reconocido")
            while flag:
                print("Que desea modificar ")
                print("1. Peso ")
                print("2. Horario de alimentacion")
                print("3. Vacunas del animal")
                print("4. Salir")
                opt=int(input())

                if opt==1:
                    print("Ingrese el peso del animal: ")
                    Animal.Animal.weight=input()
                   
                elif opt==2:
                    print("Ingrese el horario de alimentacion del animal: ")
                    Animal.Animal.feeding_schedule=input()
                
                elif opt==3:
                    print("Ingrese la actualizacion de vacunas del animal: ")
                    Animal.Animal.is_vaccinated=input()
                
                elif opt==4:
                    flag=False
                
                else:
                    print("Opcion invalida")
    
    def delete_client(self):
        print("==Ha seleccionado: eliminar un cliente==")
        Zoologic.print_clients(self)
        flag=True
        while flag==True:
            print("Que cliente desea eliminar? (Ingrese el ID del cliente a eliminar)")
            id=input()
            
            if FindID.FindID.clientName(Zoologic.clients,id)!= "incorrect":
                print("Ha seleccionado a: ",FindID.FindID.clientName(Zoologic.clients,id))
            else:
                print("ID no reconocido")
            print("Esta seguro? (S/N)")
            delete=input()
            if delete=='S' or delet=='s':
                print("Se ha eliminado a: ",FindID.FindID.clientName(Zoologic.clients,id))
                Zoologic.clients.remove(FindID.FindID.clientName(Zoologic.clients,id))
            else:
                print("No se elimino ningun cliente del sistema ")

    def delete_employee(self):
        print("==Ha seleccionado: eliminar un empleado==")
        Zoologic.print_employees(self)
        flag=True
        while flag==True:
            print("Que empleado desea eliminar? (Ingrese el ID del empleado a eliminar)")
            id=input()
            
            if FindID.FindID.employeeID(Zoologic.employees,id)!= "incorrect":
                print("Ha seleccionado a: ",FindID.FindID.employeeID(Zoologic.employees,id))
            else:
                print("ID no reconocido")
            print("Esta seguro? (S/N)")
            delete=input()
            if delete=='S' or delet=='s':
                print("Se ha eliminado a: ",FindID.FindID.employeeID(Zoologic.employees,id))
                Zoologic.employees.remove(FindID.FindID.employeeID(Zoologic.employees,id))
            else:
                print("No se elimino ningun empleado del sistema ")
    def delete_animal(self):
        print("==Ha seleccionado: eliminar un animal==")
        Zoologic.print_animals(self)
        flag=True
        while flag==True:
            print("Que animal desea eliminar? (Ingrese el ID del animal a eliminar)")
            id=input()
            
            if FindID.FindID.animalExists(Zoologic.animals,id)!= "incorrect":
                print("Ha seleccionado al animal: ",FindID.FindID.animalExists(Zoologic.animals,id))
            else:
                print("ID no reconocido")
            print("Esta seguro? (S/N)")
            delete=input()
            if delete=='S' or delet=='s':
                print("Se ha eliminado al animal: ",FindID.FindID.animalExists(Zoologic.animals,id))
                self.animals.remove(FindID.FindID.animalExists(Zoologic.animals,id))
            else:
                print("No se elimino ningun animal del sistema ")

